src/ol/interaction/translate.js

```diff
@@ -234,6 +240,35 @@ ol.interaction.Translate.prototype.setHitTolerance = function(hitTolerance) {
 
 
 /**
+ * @inheritDoc
+ */
+ol.interaction.Translate.prototype.setMap = function(map) {
+  var oldMap = this.getMap();
+  ol.interaction.Pointer.prototype.setMap.call(this, map);
+  this.updateState_(oldMap);
+};
+
+
+/**
+ * @param {ol.Map} oldMap Old map.
+ * @private
+ */
+ol.interaction.Translate.prototype.updateState_ = function(oldMap) {
```

Comment:
As an alternative, you could override `interaction.setActive()` and this could call `interaction.updateState_()` with the appropriate arguments (just as you have done with `interaction.setMap()`).

Or, if you want to continue registering a listener for change events, add a listener that expects an `ol.Object.Event` and then call `interaction.updateState_()` with the appropriate arguments.

Overriding `setActive` feels more straightforward to me.